### PR TITLE
refactor(Camera): Allow setting key matrices directly

### PR DESCRIPTION
**Pull request by sankhesh**
This change provides the ability to set the camera key matrices directly
from an external application. This ability opens up the posibility of
integrating vtk-js within a larger WebGL application.

### CODE REVIEW

**Comment by sankhesh** [comment]
@martinken @aashish24 Please review

File: Sources/Rendering/Core/Camera/index.js [hunk 1]
```diff
@@ -417,6 +417,10 @@ function vtkCamera(publicAPI, model) {
     // invert to get hmd -> world
     mat4.invert(viewMatrix, viewMatrix);
 
+    publicAPI.computeViewParametersFromViewMatrix();
+  };
+
+  publicAPI.computeViewParametersFromViewMatrix = () => {
     // then extract the params position, orientation
     // push 0,0,0 through to get a translation
     vec3.transformMat4(tmpvec1, origin, viewMatrix);
```
Comment (jourdain) [inline Sources/Rendering/Core/Camera/index.js]: viewMatrix is not defined here? You may need to store / retrieve it from the model...

File: Sources/Rendering/Core/Camera/index.js [hunk 2]
```diff
@@ -417,6 +417,10 @@ function vtkCamera(publicAPI, model) {
     // invert to get hmd -> world
     mat4.invert(viewMatrix, viewMatrix);
```
Comment (jourdain) [inline Sources/Rendering/Core/Camera/index.js]: need to save viewMatrix to model...

File: Sources/Rendering/Core/Camera/index.js (hunk 1 above)
Comment (sankhesh) [reply Sources/Rendering/Core/Camera/index.js]: Same as above.

The idea is to have a function scope viewMatrix as well as a model scope one. When an external application sets the viewMatrix explicitly, it is assigned to the model scope and used whenever `getViewMatrix` is called.

File: Sources/Rendering/Core/Camera/index.js (hunk 2 above)
Comment (sankhesh) [reply Sources/Rendering/Core/Camera/index.js]: Nope, that's intentional.
`viewMatrix` is defined in the vtkCamera function. 
See https://github.com/sankhesh/vtk-js/blob/00c21d6f77aa9c82cd97e5053fa2508c1eb9912e/Sources/Rendering/Core/Camera/index.js#L29

**Comment by jourdain** [comment]
I see thanks for the explanation. Glancing from the diff, it was not clear where that viewMatrix was defined.

File: Sources/Rendering/Core/Camera/index.js (hunk 1 above)
Comment (jourdain) [reply Sources/Rendering/Core/Camera/index.js]: Should the publicAPI.computeViewParametersFromViewMatrix function put on the public API?

**Comment by martinken** [comment]
Seems reasonable +1

File: Sources/Rendering/Core/Camera/index.js (hunk 1 above)
Comment (jourdain) [reply Sources/Rendering/Core/Camera/index.js]: Just don't bind that function on publicAPI and that will be fine. that way the function will be private inside that scope.

File: Sources/Rendering/Core/Camera/index.js [hunk 3]
```diff
@@ -438,24 +424,37 @@ function vtkCamera(publicAPI, model) {
...
   publicAPI.getViewMatrix = () => {
-    const result = mat4.create();
     if (model.viewMatrix) {
-      mat4.copy(result, model.viewMatrix);
-      mat4.transpose(result, result);
-      return result;
+      return model.viewMatrix;
     }
 
+    const result = mat4.create();
```
Comment (jourdain) [inline Sources/Rendering/Core/Camera/index.js]: maybe not needed anymore?

**Comment by sankhesh** [comment]
@jourdain Thanks for reviewing and merging 👍

**Pull request merged**